### DISCUSSION TITLE
Default layout briefly visible when using client side route guard middleware

### DISCUSSION THREAD

**Discussion by steffenstolze** [nuxt/nuxt#16723] [Questions]
Hey folks!

I have two layouts, `auth` (for everything auth-related) and `default` (for all other routes).

This is my pages folder:

<img width="428" alt="image" src="https://user-images.githubusercontent.com/8490779/192159382-bab1b360-8fcc-47c6-81b0-873cc7e16b47.png">

I have the following client middleware in place in `/middleware/auth.ts`

```typescript
export default defineNuxtRouteMiddleware((to, _from) => {
  const user = useSupabaseUser();

  // Redirect anonymous user to /auth/signin if user tries to access anything but auth pages
  if (!user.value && !to.path.includes('/auth/')) {
    return navigateTo('/auth/signin');
  }

  // /auth/signin and /auth/signup not reachable for already logged in users
  if (user.value && (to.path === '/auth/signin' || to.path === '/auth/signup')) {
    return navigateTo('/');
  }
});
```

When I access the page at root, e.g. `http://localhost:3000`, the browser loads the `pages/index.vue` that contains

```typescript
<script setup>
definePageMeta({
  middleware: 'auth',
});
</script>
```

The `auth` middleware does its job and redirects to `/auth/signin` **but not** without a short preview of the `/` route, displaying `pages/index.vue` with the default layout.
The slower the connection, the longer the `/` route with default layout is displayed, before navigating to `/auth/signin`

I don't want users to ever see the default layout and associated routes when they are not logged in, not even for half a second.

Tbh, I don't understand **why** the default layout is loaded at all, since I assumed that the `auth` middleware would prevent loading the route.

The app is configured as SPA

```javascript
export default defineNuxtConfig({
  ssr: false,
  modules: ['@nuxtjs/tailwindcss', '@nuxtjs/supabase'],
  build: {
    transpile: ['@heroicons/vue'],
  },
});
```

thanks!

**Comment by steffenstolze** [comment]
Additional info: this only happens in SPA mode (`ssr: false`). When using SSR, the redirect is done server-side and it works as expected. Thing is, I want a SPA :/

**Reply by kylemilloy** [reply]
+1. 

I seem to remember this wasn't an issue with Nuxt 2...also you could apply middleware to a layout and every subsequent page that inherited it would abide by that layout's middleware. If I'm doing something wrong, lemme know, otherwise I'd be more than happy to write a PR for that as layout level middleware is a huge plus for me and eliminates a bunch of boiler plate when 95% of my app would sit behind an auth guard.
